src/qat/purr/compiler/instructions.py

```diff
@@ -203,6 +203,11 @@ def __add__(self, other):
         new_sync.add_channels(other_targets)
         return new_sync
 
+    def __iadd__(self, other):
+        other_targets = other.quantum_targets if isinstance(other, Synchronize) else other
+        self.add_channels(other_targets)
```

Comment:
Why is this different than add? Why does it reinit sync and this doesn't?

Reply:
`__iadd__` is adding to self, `__add__` adds 2 existing instances but returns a new item that is the sum of them.

Reply:
ah ok thanks! So the other add could have just reused a sync